aa/pb.py

```diff
@@ -295,13 +296,79 @@ class PbFileFetcher(fetcher.Fetcher):
     def __init__(self, root):
         self._root = root
 
-    def _get_pb_file(self, pv, year):
-        # Split PV on either dash or colon
+    def _get_all_pb_files_of_pv(self, pv):
+        """Returns a list of all .pb files of the given PV which are found under the
+        root directory. Assuming the storage is split into STS/MTS/LTS."""
+        # Strip any STS/MTS/LTS from the root first to ensure backwards compatibility.
+        if os.path.basename(os.path.normpath(self._root)) in ["STS", "MTS", "LTS"]:
+            self._root = os.path.dirname(os.path.normpath(self._root))
+        pv_files = []
+        # Split PV on either dash or colon.
         parts = re.split("[-:]", pv)
         suffix = parts.pop()
-        directory = os.path.join(self._root, os.path.sep.join(parts))
-        filename = "{}:{}.pb".format(suffix, year)
-        return os.path.join(directory, filename)
+        # Go through long/mid/short term storages and collect all files for this PV.
+        for s in ["LTS", "MTS", "STS"]:
+            directory = os.path.join(self._root, s, os.path.sep.join(parts))
+            for f in sorted(glob.glob(os.path.join(directory, f"{suffix}*pb"))):
+                pv_files.append(f)
+        return pv_files
+
+    def _create_datetime_for_pb_file(self, filepath):
+        """Each .pb files of the archiver appliance ends with a date information
+        corresponding to the stored data. This function returns a datetime.datetime
+        object matching the date information of the given .pb file."""
+        filename = os.path.basename(filepath)
+        # The filename can contain just the year or stepwise more information up to
+        # year, month, day, hour and minutes.
+        dates = (
+            re.search(r"\d{4}(_\d{2})?(_\d{2})?(_\d{2})?(_\d{2})?", filename)
+            .group(0)
+            .split("_")
+        )
+        dates = [int(date) for date in dates]
+        # Make sure to give a least 3 arguments to datetime.datetime.
+        while len(dates) < 3:
+            dates.append(1)
+        return datetime.datetime(*dates, tzinfo=pytz.utc)
```

Comment:
It would be good to guard against malformed input here too. For example, if one of the integers is out of range for a day / hour, you'll probably get an exception from this line.